CAP/lib/CAP/Model/Portals.pm

```diff
@@ -51,13 +51,15 @@ sub BUILD {
 sub portal_from_host {
   my ( $self, $host ) = @_;
 
-  my $subd = substr( $host, 0, index( $host, '.' ) );
-  if ( index( $subd, '-' ) > -1 ) {
-    $subd = substr( $subd, 0, index( $subd, '-' ) );
-  }
+  if( defined $host ) {
```

Comment:
Remember that you have many choices, depending on which format you consider more readable.

`return if (! defined $host);`